source/manual/zendesk-smartsurvey-feedback-data-pipeline.html.md.erb

```diff
@@ -0,0 +1,145 @@
+
+## Copy and cleanse the feedback data
+
+SmartSurvey and Zendesk store their feedback data on their own servers.
+
+As this feedback data is stored in two different servers, accessing and using the data is difficult.
+
+The following process allows GDS users to access this data more easily.
+
+Every day at approximately 8:10am, an AWS Lambda function runs a collection of Python scripts to copy the feedback data from the SmartSurvey and Zendesk servers to BigQuery.
+
+These Python scripts are located in the [govuk-user-journey-models GitHub repo on alphagov](https://github.com/alphagov/govuk-user-journey-models).
+
+To get access to this repo, speak to the GDS Data Labs team on the [#govuk-data-labs Slack channel](https://app.slack.com/client/T8GT9416G/CHR4UQKU4).
+
+Whilst the data is being copied to BigQuery, the Python scripts process and cleanse the data to mask and remove personally identifiable information (PII). This is a 2 step process.
+
+1. The scripts use regex matching to look for National Insurance, NHS and credit card numbers. The scripts then mask this PII. The regex patterns are as follows:
+
+    ```
+    [r'(?i)\b(?:[a-z]\W?){2}(?:\d\W?){6}\W?[a-z]\b', '[NI###]'],
+    [r'(?i)nhs\s?(?:.{0,20})?(?:\d\D?){9}(?:\d)', '[NHS###]'],
+    [r'(?:\d\D?){16}', '[CC###]']
+    ```
+
+1. The scripts send the data to the [Google Cloud Data Loss Prevention service (DLP)](https://cloud.google.com/dlp). Google Cloud DLP looks for the following information types:
+
+    ```
+    PERSON_NAME,
+    EMAIL_ADDRESS",
+    DATE_OF_BIRTH,
+    STREET_ADDRESS
+    ```
+
+Google Cloud DLP then cleanses the data of these information types and returns the cleansed data to BigQuery.
+
+## Save the feedback data in BigQuery
+
+When the data cleansing is complete, the data is saved into the GOV.UK Analytics project in BigQuery, `govuk-bigquery-analytics`.
+
+Within this project:
+
+- the Zendesk data is stored in the `zendeskdata` data set
+- the SmartSurvey data is stored in the `UISData` data set
+
+Within these data sets are the daily data tables. Each day’s data is created as a new data table with a `YYYYMMDD` format. For example, for 9 March 2021:
+
+- the SmartSurvey data table is `UISData_20210309`
+- the Zendesk data table is `zendesk_20210309`
+
+To request permission to view or edit these data tables in BigQuery, speak to the GDS Data Labs team on the [#govuk-data-labs Slack channel](https://app.slack.com/client/T8GT9416G/CHR4UQKU4).
+
+The Zendesk and SmartSurvey feedback data is now in BigQuery, ready for users to view and analyse.
+
+See the [documentation on presenting or extract feedback data from BigQuery](/manual/view-extract-feedback-data-bigquery.html) for more information on the next step of the process.
+
+## Data field definitions
+
+All data fields in BigQuery are text strings.
+
+### SmartSurvey data fields
+
+|Data field|Definition|
+|:---|:---|
+|content|content|
+|content|content|
+|UserID|TBC|
```

Comment:
TBC? to be confirmed?

Reply:
correct - when I drafted this content with Ian, we didn't have descriptions of all the data fields. Does this information exist anywhere?